multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -0,0 +1,1715 @@
+from __future__ import annotations
```

Comment:
if everything about matcher is moved here, should we delete the matcher related code in the predictor.py ?

Reply:
Good point. Removed the matcher related in predictor.py.